applications/tari_validator_node/src/p2p/services/mempool/initializer.rs

```diff
@@ -20,22 +20,25 @@
 //  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 //  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use tari_comms::types::CommsPublicKey;
 use tari_dan_engine::instruction::Transaction;
-use tokio::{sync::mpsc, task};
+use tokio::{
+    sync::{broadcast, mpsc},
+    task,
+};
 
 use crate::p2p::services::{
     mempool::{handle::MempoolHandle, service::MempoolService},
     messaging::OutboundMessaging,
 };
 
 pub fn spawn(
-    new_transactions: mpsc::Receiver<(CommsPublicKey, Transaction)>,
+    new_transactions: mpsc::Receiver<Transaction>,
+    new_transactions_sender: mpsc::Sender<Transaction>,
     outbound: OutboundMessaging,
 ) -> MempoolHandle {
-    let (tx_valid_transactions, rx_valid_transactions) = mpsc::unbounded_channel();
+    let (tx_valid_transactions, rx_valid_transactions) = broadcast::channel(100);
```

Comment:
Only concern here is that we will drop transactions if we cant process them fast enough and we should account for a massive amount of transactions on the 2nd layer. We should probably separate the inbound tx channel from the handle because (1) there is only going to be one receiver of transactions and (2) so that the mempool handle can be copied.

Reply:
Yeah was also thinking this